controllers/product_controllers.js

```diff
@@ -78,3 +78,67 @@ module.exports.getVariantInfo = async (req, res, next) => {
         res.sendStatus(500)
     }
 }
+
+module.exports.updateCrudProduct = async (req, res, next) => {
+    try {
+        const  _id = req._id;
+        const {name, description, section, brand} = req.body;
+        const user = await Product.findOneAndUdpate(_id, name, description, section, brand);
+        res.status(200).json(user);
+    } catch (error) {
+        res.sendStatus(500) && next(error);
+    }
+}
+
+module.exports.removeCrudProduct = async (req, res, next) => {
+    try {
+        const  _id = req._id;
+        const {name, description, section, brand} = req.body;
+        const user = await Product.prodRemove(_id, name, description, section, brand);
```

Comment:
En este usarias Product.findByIdAndDelete():
https://mongoosejs.com/docs/api/model.html#Model.findByIdAndDelete()
Tambien tendiras que asegurarte de borrar todos las variantes que hagan referencia a ese producto (Como cascade en sql).
Habria que implement algo asi en 'models/product.js'
// Define the 'pre' hook for 'remove' method of Product model
productSchema.pre('remove', async function(next) {
  try {
    // Delete all variants associated with this product
    await Variant.deleteMany({ product: this._id });
    next();
  } catch (error) {
    next(error);
  }
});

Enfocate en las demas y deja esta para lo ultimo cuando ya tengas mas experiencia. Despues la miramos juntos si te parece dificil.